src/app/views/query-response/graph-toolkit/GraphToolkit.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { Component } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { connect } from 'react-redux';
+
+import { Label } from 'office-ui-fabric-react';
+import { IQuery } from '../../../../types/query-runner';
+import { parseSampleUrl } from '../../../utils/sample-url-generation';
+
+const templateMap: any = {
+  '/me/planner/tasks' : 'https://mgt.dev/iframe.html?id=components-mgt-tasks--tasks',
+  '/me': 'https://mgt.dev/iframe.html?id=components-mgt-person-card--person-card-hover',
+  '/me/people' : 'https://mgt.dev/iframe.html?id=components-mgt-people--people',
+};
+
+class GraphToolkit extends Component<any> {
+  constructor(props: any) {
+    super(props);
+  }
+
+  public lookupToolkitUrl = (sampleQuery: IQuery): any => {
+    if (sampleQuery) {
+      const { requestUrl, search } = parseSampleUrl(sampleQuery.sampleUrl);
+      const query = requestUrl + search;
+      // find if the url of the request has a template mapped to it
+      for (const templateMapKey in templateMap) {
+        if (templateMap.hasOwnProperty(templateMapKey)) {
+          // check if the template matches a specific pattern while ignoring case
+          const isMatch = new RegExp(templateMapKey + '$', 'i').test('/' + query);
+          if (isMatch) {
+            return templateMap[templateMapKey];
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  public render() {
```

Comment:
Having multiple returns in the render function can get confusing pretty quick.

I suggest we create two functions and call them in the render function.

```javascript
private showToolkit() {...}
private showLable() {...}

render() {
   if (toolkitUrl) {
       this.showToolkit()
   } else {
       this.showLabel()
}
```

Reply:
This would be necessary if we had more than just the two return statements.

Reply:
It doesn't hurt to do it earlier.